### PR TITLE
MWPW-155315 Set Marketo POI

### PR DESCRIPTION
**Pull request by Brandon32**
* Set Marketo POI based on session storage or query parameter.

Resolves: [MWPW-155315](https://jira.corp.adobe.com/browse/MWPW-155315)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/bmarshal/marketo/entry?martech=off
- After: https://bmarshal-marketo-poi--milo--adobecom.hlx.page/drafts/bmarshal/marketo/entry?martech=off

**BACOM URLs:**
- Before: https://main--bacom--adobecom.hlx.live/drafts/bmarshal/marketo/poi/entry?martech=off
- After: https://bmarshal-marketo-poi--bacom--adobecom.hlx.live/drafts/bmarshal/marketo/poi/entry?martech=off

### CODE REVIEW

File: libs/blocks/marketo/marketo.js [hunk 1]
```diff
@@ -39,6 +41,9 @@ const FORM_MAP = {
   'sfdc-campaign-id': 'program.campaignids.sfdc',
 };
```
Comment (meganthecoder) [inline libs/blocks/marketo/marketo.js:43]: Don't need an extra line between consts

File: libs/blocks/marketo/marketo.js [hunk 2]
```diff
@@ -123,6 +128,21 @@ export const formSuccess = (formEl, formData) => {
...
 
+export function setProductOfInterest(formData, search = window.location.search) {
+  if (formData[PROGRAM_POI]) return;
+
+  const queryPoi = new URLSearchParams(search).get(QUERY_POI);
+  if (queryPoi === 'clear') {
+    sessionStorage.removeItem(SESSION_POI);
+    return;
+  }
```
Comment (meganthecoder) [inline libs/blocks/marketo/marketo.js:138]: Clearing and returning without setting a poi seems a little strange. Is this just for dev purposes?

Comment (Brandon32) [reply libs/blocks/marketo/marketo.js:138]: Yes, this was for dev purposes, I can remove this.

**Comment by Brandon32** [comment]
> Couple small comments. Also, you've got some eslint and nala errors

Fixed eslint and nala errors, project code coverage appears to be bugged.

File: libs/blocks/marketo/marketo.js [hunk 3]
```diff
@@ -123,6 +127,16 @@ export const formSuccess = (formEl, formData) => {
...
 };
 
+export function setProductOfInterest(formData, search = window.location.search) {
+  if (formData[PROGRAM_POI]) return;
+
+  const queryPoi = new URLSearchParams(search).get(QUERY_POI);
+  const sessionPoi = sessionStorage.getItem(SESSION_POI);
+  const productOfInterest = queryPoi || sessionPoi || null;
+
+  formData[PROGRAM_POI] = productOfInterest;
```
Comment (JasonHowellSlavin) [inline libs/blocks/marketo/marketo.js:137]: Is it alright to set null in this case?

File: test/blocks/marketo/marketo.test.js [hunk 4]
```diff
@@ -42,6 +43,41 @@ describe('marketo', () => {
   });
 });
 
+describe('marketo setProductOfInterest', () => {
+  beforeEach(() => {
+    sinon.stub(sessionStorage, 'getItem');
```
Comment (JasonHowellSlavin) [inline test/blocks/marketo/marketo.test.js:48]: since wtr runs everything in a browser, is there a reason you are not just using sessionStorage?

**Review by JasonHowellSlavin — APPROVED** [review]

File: libs/blocks/marketo/marketo.js (hunk 3 above)
Comment (Brandon32) [reply libs/blocks/marketo/marketo.js:137]: Thanks, `productOfInterest` will already be null, I should only set `formData` if `productOfInterest` is truthy.

File: test/blocks/marketo/marketo.test.js (hunk 4 above)
Comment (Brandon32) [reply test/blocks/marketo/marketo.test.js]: True, updated

**Pull request closed without merging**